chain/network/src/routing_table_actor.rs

```diff
@@ -14,22 +16,355 @@ use crate::ibf::{Ibf, IbfBox};
+        });
+
+        // Update metrics after edge update
+        near_metrics::inc_counter_by(&metrics::EDGE_UPDATES, total as u64);
+        near_metrics::set_gauge(&metrics::EDGE_ACTIVE, self.raw_graph.total_active_edges as i64);
+
+        edges
+    }
+
+    /// If peer_id is not in memory check if it is on disk in bring it back on memory.
+    fn fetch_edges_for_peer_from_disk(&mut self, other_peer_id: &PeerId) {
+        if other_peer_id == self.my_peer_id()
+            || self.peer_last_time_reachable.contains_key(other_peer_id)
+        {
+            return;
+        }
+
+        let my_peer_id = self.my_peer_id().clone();
+
+        // Get the "row" (a.k.a nonce) at which we've stored a given peer in the past (when we pruned it).
+        if let Ok(component_nonce) = self.component_nonce_from_peer(other_peer_id.clone()) {
+            let mut update = self.store.store_update();
+
+            // Load all edges that were persisted in database in the cell - and add them to the current graph.
+            if let Ok(edges) = self.get_and_remove_component_edges(component_nonce, &mut update) {
+                for edge in edges {
+                    for &peer_id in vec![&edge.peer0, &edge.peer1].iter() {
+                        if peer_id == &my_peer_id
+                            || self.peer_last_time_reachable.contains_key(peer_id)
+                        {
+                            continue;
+                        }
+
+                        // We remove mapping from `peer_id` to component_nonce else if
+                        // the mapping from `peer_id`, that we store in DB, points to the current nonce.
+                        if let Ok(cur_nonce) = self.component_nonce_from_peer(peer_id.clone()) {
+                            if cur_nonce == component_nonce {
+                                self.peer_last_time_reachable
+                                    .insert(peer_id.clone(), Instant::now() - SAVE_PEERS_MAX_TIME);
+                                update
+                                    .delete(ColPeerComponent, Vec::from(peer_id.clone()).as_ref());
+                            }
+                        }
+                    }
+                    self.add_verified_edge(edge);
+                }
+            }
+
+            if let Err(e) = update.commit() {
+                warn!(target: "network", "Error removing network component from store. {:?}", e);
+            }
+        } else {
+            self.peer_last_time_reachable.insert(other_peer_id.clone(), Instant::now());
+        }
+    }
+
+    fn my_peer_id(&self) -> &PeerId {
+        &self.raw_graph.source
+    }
+
+    /// Recalculate routing table and update list of reachable peers.
+    /// If pruning is enabled we will remove unused edges and store them to disk.
+    ///
+    /// # Returns
+    /// List of edges removed.
+    pub fn recalculate_routing_table(
+        &mut self,
+        prune: Prune,
+        prune_edges_after: Duration,
```

Comment:
could we add this prune_edges_after to 'Allow' enum from Prune?

Reply:
Well, the name is confusing. We prune edges that were not rechable for `prune_edges_after` duration.
`Force` is used for testing, it runs pruning every time.
`Allow` only does pruning at most 1 per hour.

I renamed `prune_edges_after` to `prune_edges_not_reachable_for`.